spatialdata/_dl/datasets.py

```diff
@@ -0,0 +1,130 @@
+from typing import Callable, Optional
```

Comment:
nitpick: can we call this module different than _dl but something like `_dataloader` (although also not great). Mostly asking cause this could also be used to simply create a dataloader to then e./g. compute features on tiles, similar to https://squidpy.readthedocs.io/en/latest/api/squidpy.im.calculate_image_features.html#squidpy.im.calculate_image_features

Reply:
renamed